src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,207 @@
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAnglePosition().getDegrees(), getAngleVelocityDegreesPerSecond())
+        );
+
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void generateElevatorMotorProfile(double targetElevatorPosition, double speedPercentage) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ELEVATOR_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetElevatorPosition, 0),
+                new TrapezoidProfile.State(getElevatorPositionRevolutions(), getElevatorVelocityRevolutionsPerSecond())
+        );
+
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private double calculateAngleMotorOutput(TrapezoidProfile.State targetState) {
+        double pidOutput = ArmConstants.ANGLE_PID_CONTROLLER.calculate(
+                getAnglePosition().getDegrees(),
+                targetState.position
+        );
+        double feedforward = ArmConstants.ANGLE_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                Units.degreesToRadians(targetState.velocity)
+        );
+
+        return pidOutput + feedforward;
+    }
+
+    private double calculateElevatorMotorOutput(TrapezoidProfile.State targetState) {
+        double pidOutput = ArmConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                getElevatorPositionRevolutions(),
+                targetState.position
+        );
+        double feedforward = ArmConstants.ELEVATOR_FEEDFORWARD.calculate(targetState.velocity);
+        return pidOutput + feedforward;
+    }
+
+    private double getAngleMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastAngleMotorProfileGenerationTime;
+    }
+
+    private double getElevatorMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastElevatorMotorProfileGenerationTime;
+    }
+
+    private Rotation2d getAnglePosition() {
+        double positionRevolutions = ArmConstants.ANGLE_MOTOR_POSITION_SIGNAL.refresh().getValue();
+        return Rotation2d.fromRotations(positionRevolutions);
+    }
+
+    private double getElevatorPositionRevolutions() {
+        return Conversions.magTicksToRevolutions(elevatorEncoder.getSelectedSensorPosition());
+    }
+
+    private double getAngleVelocityDegreesPerSecond() {
+        double positionRevolutions = ArmConstants.ANGLE_MOTOR_VELOCITY_SIGNAL.refresh().getValue();
+        return Conversions.revolutionsToDegrees(positionRevolutions);
+    }
+
+    private double getElevatorVelocityRevolutionsPerSecond() {
+        double magTicksPerSecond = Conversions.perHundredMsToPerSecond(elevatorEncoder.getSelectedSensorVelocity());
+        return Conversions.magTicksToRevolutions(magTicksPerSecond);
+    }
```

Comment:
1. You can't mention hardware
2. No need for these functions anymore (apart from the `getAnglePosition` one) since they can just be passed with correct units in the inputs class